src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+        using utils::read_json_param;
+
+        set_scheduler(Scheduler::from_config(root_dir + "/scheduler/scheduler_config.json"));
+
+        const std::string text_encoder = data["text_encoder"][1].get<std::string>();
+        if (text_encoder == "CLIPTextModelWithProjection") {
+            m_clip_text_encoder_1 =
+                std::make_shared<CLIPTextModelWithProjection>(root_dir + "/text_encoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        const std::string text_encoder_2 = data["text_encoder_2"][1].get<std::string>();
+        if (text_encoder_2 == "CLIPTextModelWithProjection") {
+            m_clip_text_encoder_2 =
+                std::make_shared<CLIPTextModelWithProjection>(root_dir + "/text_encoder_2", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        // TODO: text_encoder_3
+
+        const std::string vae = data["vae"][1].get<std::string>();
+        if (vae == "AutoencoderKL") {
+            m_vae_decoder = std::make_shared<AutoencoderKL>(root_dir + "/vae_decoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", vae, "' VAE decoder type");
+        }
+
+        const std::string transformer = data["transformer"][1].get<std::string>();
+        if (transformer == "SD3Transformer2DModel") {
+            m_transformer = std::make_shared<SD3Transformer2DModel>(root_dir + "/transformer", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", transformer, "'Transformer type");
+        }
+
+        // initialize generation config
+        initialize_generation_config(data["_class_name"].get<std::string>());
+    }
+
+    StableDiffusion3Pipeline(const CLIPTextModelWithProjection& clip_text_model_1,
+                             const CLIPTextModelWithProjection& clip_text_model_2,
+                             const SD3Transformer2DModel& transformer,
+                             const AutoencoderKL& vae_decoder)
+        : m_clip_text_encoder_1(std::make_shared<CLIPTextModelWithProjection>(clip_text_model_1)),
+          m_clip_text_encoder_2(std::make_shared<CLIPTextModelWithProjection>(clip_text_model_2)),
+          m_vae_decoder(std::make_shared<AutoencoderKL>(vae_decoder)),
+          m_transformer(std::make_shared<SD3Transformer2DModel>(transformer)) {}
+
+    void reshape(const int num_images_per_prompt,
+                 const int height,
+                 const int width,
+                 const float guidance_scale) override {
+        check_image_size(height, width);
+
+        const size_t batch_size_multiplier =
+            do_classifier_free_guidance(guidance_scale) ? 2 : 1;  // Unet accepts 2x batch in case of CFG
+        m_clip_text_encoder_1->reshape(batch_size_multiplier);
+        m_clip_text_encoder_2->reshape(batch_size_multiplier);
```

Comment:
batch_size_multiplier = 2 for CFG implies you call text encoder with both negative and positive prompt at the same time and do_classifier_free_guidance=True

Reply:
updated `reshape` method